### PR TITLE
Add exchanges tests

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
requires https://github.com/Drakkar-Software/OctoBot-Trading/pull/778
requres https://github.com/Drakkar-Software/OctoBot-Tentacles/pull/749

### CODE REVIEW

File: exchanges_tests/test_phemex.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    async def test_get_closed_orders(self):
+        await super().test_get_closed_orders()
+
+    async def test_create_and_cancel_limit_orders(self):
+        # 08 dec 2022: did not run so far, testnet is bugged: order are apparently accepted but never show up
+        await super().test_create_and_cancel_limit_orders()
+
+    async def test_create_and_fill_market_orders(self):
+        # 08 dec 2022: did not run so far, testnet is bugged: order are apparently accepted but never show up
```
Comment (Herklos) [inline exchanges_tests/test_phemex.py:48]: 👍

File: exchanges_tests/test_bitget.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    abstract_authenticated_exchange_tester.AbstractAuthenticatedExchangeTester
+):
+    # enter exchange name as a class variable here
+    EXCHANGE_NAME = "bitget"
+    ORDER_CURRENCY = "BTC"
+    SETTLEMENT_CURRENCY = "USDT"
+    SYMBOL = f"{ORDER_CURRENCY}/{SETTLEMENT_CURRENCY}"
+    ORDER_SIZE = 40  # % of portfolio to include in test orders
+    CONVERTS_ORDER_SIZE_BEFORE_PUSHING_TO_EXCHANGES = True
+    IGNORE_ORDER_FEE = True
```
Comment (Herklos) [inline exchanges_tests/test_bitget.py:36]: Why should we ignore fee here?

File: exchanges_tests/abstract_authenticated_future_exchange_tester.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+                                                             symbol=symbol)
+        return await super().create_market_stop_loss_order(
+            current_price, stop_price, size, side, symbol=symbol,
+            push_on_exchange=push_on_exchange
+        )
+
+    async def create_order(self, price, current_price, size, side, order_type,
+                           symbol=None, push_on_exchange=True):
+        # contracts are required to create orders
+        await self.load_contract(symbol)
```
Comment (Herklos) [inline exchanges_tests/abstract_authenticated_future_exchange_tester.py:115]: 👍

File: exchanges_tests/abstract_authenticated_exchange_tester.py [hunk 4]
```diff
@@ -0,0 +1,538 @@
...
+    async def wait_for_cancel(self, order):
+        await self._get_order_until(order, personal_data.parse_is_cancelled, self.CANCEL_TIMEOUT)
+
+    async def wait_for_edit(self, order, edited_quantity):
+        def is_edited(row_order):
+            return decimal.Decimal(str(row_order[trading_enums.ExchangeConstantsOrderColumns.AMOUNT.value])) \
+                   == edited_quantity
+        await self._get_order_until(order, is_edited, self.EDIT_TIMEOUT)
+
+    async def _get_order_until(self, order, validation_func, timeout):
```
Comment (Herklos) [inline exchanges_tests/abstract_authenticated_exchange_tester.py:485]: 👍

File: exchanges_tests/test_bitget.py (hunk 2 above)
Comment (GuillaumeDSM) [reply exchanges_tests/test_bitget.py:36]: fees are not always provided by the exchange

**Review by Herklos — APPROVED** [review]
👍

**Review by Herklos — APPROVED** [review]

**Comment by GuillaumeDSM** [comment]
replaced by https://github.com/Drakkar-Software/OctoBot/pull/2133

**Pull request closed without merging**